File: Facade/Golang/HotelKeeperFacade.go
package main

type HotelKeeperFacade struct {}

func (hotelKeeperFacade *HotelKeeperFacade) getVegMenu()  {
	vegRestaurant := NewVegRestaurant()

	vegRestaurant.getMenus().describe()
}

func (hotelKeeperFacade *HotelKeeperFacade) getNonVegMenu()  {
	nonVegRestaurant := NewNonVegRestaurant()

	nonVegRestaurant.getMenus().describe()
}

func (hotelKeeperFacade *HotelKeeperFacade) getVegNonMenu()  {
	vegNonRestaurant := NewVegNonRestaurant()

	vegNonRestaurant.getMenus().describe()
}

File: Facade/Python 3/HotelKeeperFacade.py
from NonVegRestaurant import NonVegRestaurant
from VegNonBothRestaurant import VegNonBothRestaurant
from VegRestaurant import VegRestaurant

class HotelKeeperFacade:
  def getVegMenu(self):
    vegRestaurant = VegRestaurant()
    vegRestaurant.getMenus().describe()

  def getNonVegMenu(self):
    nonVegRestaurant = NonVegRestaurant()
    nonVegRestaurant.getMenus().describe()

  def getVegNonMenu(self):
    vegNonBothRestaurant = VegNonBothRestaurant()
    vegNonBothRestaurant.getMenus().describe()
